runtime/include/cloe/data_broker.hpp

```diff
@@ -0,0 +1,1306 @@
+      const TValue* value = getter_fn->operator()();
+      return value;
+    }
+    throw std::logic_error(
+        fmt::format("Unable to get value from signal '{}' since getter-function was not provided.",
+                    names_.front()));
+  }
+  /**
+    * Returns the getter function of the signal
+    * \tparam TValue Type of the signal
+    * \return Getter function of the signal
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, const typed_set_value_function_t<TValue>*>::type setter()
+      const {
+    validate_type<TValue>();
+    const typed_set_value_function_t<TValue>* set_value_fn =
+        std::any_cast<typed_set_value_function_t<TValue>>(&set_value_);
+    return set_value_fn;
+  }
+  /**
+    * Sets the setter function of the signal
+    * \tparam TValue Type of the signal
+    * \param set_value Getter function of the signal
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, void>::type set_setter(
+      typed_set_value_function_t<TValue> set_value) {
+    validate_type<TValue>();
+    set_value_ = set_value;
+  }
+  /**
+    * Sets the value of the signal
+    * \tparam TValue Type of the signal
+    * \param value Value of the signal
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, void>::type set_value(const TValue* value) const {
+    validate_type<TValue>();
+    const typed_set_value_function_t<TValue>* setter_fn = setter<TValue>();
+    if (setter_fn && *setter_fn) {
+      setter_fn->operator()(value);
+      return;
+    }
+    throw std::logic_error(fmt::format(
+        "Unable to set value since setter-function not provided. Signal: {}", names_.front()));
+  }
+  /**
+    * Return the trigger function for the value_changed event
+    * \tparam TValue Type of the signal
+    * \return Trigger function for raising the value_changed event
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, const typed_on_value_change_event_function_t<TValue>&>::type
+  trigger() const {
+    validate_type<TValue>();
+    const typed_on_value_change_event_function_t<TValue>* trigger_fn =
+        std::any_cast<typed_on_value_change_event_function_t<TValue>>(&on_value_changed_);
+    assert(trigger_fn);
+    return *trigger_fn;
+  }
+
+ private:
+  /**
+    * Unpacks the event to Event<const T*> and provides it to the caller
+    * 
+    * \tparam T Type of the event
+    * \param callback Caller function which accepts the unpacked event
+    * \note: In case type T and actual type do not match an exception is thrown
+    */
+  template <typename TValue>
+  void subscribe_impl(const std::function<void(Event<const TValue*>&)>& callback) {
+    validate_type<TValue>();
+    typed_value_changed_event_t<TValue>* value_changed_event =
+        std::any_cast<typed_value_changed_event_t<TValue>>(&value_changed_event_);
+    if (callback) {
+      assert(("value_changed_event must not be nullptr", value_changed_event));

```

Comment:
I'm getting this compiler feedback:
```
../../include/cloe/data_broker.hpp:529:15: error: left operand of comma operator has no effect [-Werror=unused-value]
  529 |       assert(("value_changed_event must not be nullptr", value_changed_event));
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Above, use `assert(value_changed_event && "message")`.

I think, `assert(value_changed_event != nullptr)` is clear enough though, I don't see added value from the message. This kind of check is common enough to not need an explanation.

Also: doesn't nullptr indicate that the any_cast failed? Then a better error message would be to point out that the type is wrong.